### PR TITLE
Rewrote http.py to be faster

### PR DESCRIPTION
**Pull request by theseion**
…, mainly by using select() instead of timeouts

Split from PR #66, part 1

### CODE REVIEW

**Review by fzipi — CHANGES_REQUESTED** [review]
Hmmm...

- requirements.txt updates should be in the dependency updates only.
- same with the test updates, should be in the tests only

**Comment by theseion** [comment]
I did that so the PR checks would run through and to show, that the change doesn't affect the tests. Do you still want me to change it?

**Comment by fzipi** [comment]
Are the tests failing when upgrading dependencies?

**Comment by theseion** [comment]
Yes, partly because the domains used in the tests are unreliable.

**Comment by fzipi** [comment]
Well, then we should change the description on this PR to match whatever is doing. Also, if someone needs to revert it will erase both the http.py rewriting AND the tests update.

I think that the easiest way to solve this is to create a new PR for updating the tests and the reason of reliability. We merge that one. Then rebase this one so it doesn't update tests also. We merge this one.

Then rebase the following and keep going :)

**Comment by theseion** [comment]
That PR already exists :) Your point about reverting is a good one. I'll update the PR's and indicate the proper order in which to review them.

**Comment by theseion** [comment]
@fzipi @fgsch Final one.

File: ftw/http.py [hunk 1]
```diff
@@ -463,38 +462,77 @@ def get_response(self):
...
+                if (sys.platform == 'win32' and
+                        err.errno == errno.WSAEWOULDBLOCK):
                     pass
                 elif (self.request_object.protocol == 'https' and
-                      err.args[0] == ssl.SSL_ERROR_WANT_READ):
-                    continue
-                # If we didn't it's an error
+                        err.args[0] == ssl.SSL_ERROR_WANT_READ):
+                    ready_sock, _, _ = select.select(
+                        [self.sock], [], [self.sock], .3)
```
Comment (fzipi) [inline ftw/http.py]: In the end, this `.3` is the `self.HTTP_TIMEOUT` from above, right? Why not make it explicit? Looks really hidden here.

Comment (theseion) [reply ftw/http.py]: Yes, you're right.

**Review by fzipi — APPROVED** [review]
LGTM. @fgsch What do you think?

**Comment by fzipi** [comment]
Merging.

**Pull request merged**